department-of-veterans-affairs/caseflow
# SeanC/APPEALS-42317 | Migrate existing Supplemental Claim Remands to Remand records via script

Resolves [[APPEALS-42317] Migrate existing Supplemental Claim Remands to Remand records via script](https://jira.devops.va.gov/browse/APPEALS-42317)

# Description
Created a migration file that will back fill the new type column, based whether it is a Supplemental Claim or a Remand, for all of the records in the supplemental claims table.

## Acceptance Criteria
- [x] Create a one-time, standalone script/job/service that will:
- [x] Fill in type field for every record on the supplemental claims table
- [x] Implement ability to rollback changes made by the migration script
- [x] Verify successful execution of the script and migration to the new Remands records
- [x] Write detailed documentation for the script

## Testing Plan

### Data Setup (run these commands in the rails console):
1. (Can skip this step if you already have supplemental claims in your db)run: `FactoryBot.create_list(:supplemental_claim, 5)`
2. `appeals = FactoryBot.create_list(:remand, 5)`
3. `appeals.map{|appeal| appeal.update type: "SupplementalClaim"}`
4. `sc = FactoryBot.create(:supplemental_claim)`
5. `sc.update decision_review_remanded_type: 'HigherLevelReview'`
6. `remand = FactoryBot.create(:remand)`
7. `remand.update decision_review_remanded_type: 'HigherLevelReview'`
### Test Steps:
1. Run command: `make migrate` in the terminal
2. Open the supplemental_claims table
3. Confirm that all of the records on the table have either "Supplemental Claim" or "Remand" in the type column 
8. Confirm that the view 'remands' exists and has the same number of records as supplemental claims have Remand in the type column
9. In the rails console run: `FactoryBot.create(:remand)`
10. In the rails console run: `FactoryBot.create(:supplemental_claim)`
11. Verify that the remands view has 1 more record in it

# Backend
## Database Changes

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
